docs/workspaces.md

```diff
@@ -40,6 +40,48 @@ resources](./templates.md#persistent-and-ephemeral-resources).
 
 When a workspace is deleted, all of the workspace's resources are deleted.
 
+## Workspace Schedule
+
+You can set a schedule for you workspace to automatically start at a given time (e.g. Monday - Friday at 9 AM).
+
+![Auto start](./images/auto-start.png)
+
+You can also set a "time until shutdown" to auto-stop your workspace (e.g "8" hours after stopped).
+
+![Auto stop](./images/auto-stop.png)
+
+In order to use your workspace past the "time until shutdown," you can use bumpers in the dashboard to keep the workspace alive:
+
+![Dashboard bumpers](./images/add-bumper-terminal.png)
+
+### Activity detection (alpha)
+
+To keep your workspace running past the "time until shutdown," Coder can bump the workspace lifetime by 1 hour when it detects any of the following activities:
+
+- Open tunnel (SSH connection)
+
+  If you have an open tunnel/SSH connection, the CLI will send keepalives to Coder, even if you are not typing.
+
+  Use `coder tunnel --disable-keep-alive` or `coder config-ssh --disable-keep-alive` to avoid send keepalives to Coder.
+
+- HTTP (or websocket) request on `coder_app`
+
+  Coder's proxy will detect any traffic on apps (e.g. code-server)
```

Comment:
For websockets, it is not clear to me whether we bump the lifetime only on the websocket request, or periodically while it is open like the SSH connection

Reply:
@Emyrk any thoughts here? I'd imagine websockets are more likely to be left open than SSH, but I'm admittedly not the expert here 🤷🏼

Reply:
@spikecurtis would love your thoughts too. Happy to also flip a coin and then learn from a GitHub issue we should do something differently :P

Reply:
I do know that Jupyter keeps a websocket open while you are interacting with a notebook.  So, without periodic bumps, we would not consider a user "active" if they were working in a single Jupyter notebook.  (opening new notebooks would cause a bump)

If we're going to do periodic bumps on an SSH connection, I think we should also do the same for any websockets open to coder_apps

Reply:
Makes sense to me!

Reply:
Wouldn't any ide app have some sort of long lived connection? For sending keystrokes and whatnot?

I agree with Spike